src/ansiblelint/runner.py

```diff
@@ -458,7 +453,7 @@ def _emit_matches(self, files: list[Lintable]) -> Generator[MatchError, None, No
                 except MatchError as exc:
                     if not exc.filename:  # pragma: no branch
                         exc.filename = str(lintable.path)
-                    exc.rule = LoadingFailureRule()
+                    exc.rule = self.rules.rules[1]
```

Comment:
Do not assume order of rules! If you want you might modify the getter to allow you to get a rule by its ID from self.rules, but never use an index!